src/mrinufft/operators/base.py

```diff
@@ -310,10 +311,8 @@ def make_autograd(self, variable="data"):
         """
         if not AUTOGRAD_AVAILABLE:
             raise ValueError("Autograd not available, ensure torch is installed.")
```

Comment:
This should check that the backend supports autodiff. 
Ideally, this should be a class attribute defined for each backend (by default False in the `FourierOperatorBase`. I would suggest something along these lines:

```python 
class FourierOperatorBase: 
     ... 
     autograd_available = False 

# in each backend that support autograd, e.g. 
class MRICufinufft(FourierOperatorBase):
      backend = "cufinufft"
      autograd_available = True 
```

Then, in this function we will test that both AUTOGRAD_AVAILABLE and self.autograd_available are True. Otherwise, raise an error (with a specific error message in each case)